tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml.cs

```diff
@@ -369,4 +351,45 @@ public void UpdateDevDriveInfo()
         AddRepoViewModel.FolderPickerViewModel.InDevDriveScenario = true;
         AddRepoViewModel.EditDevDriveViewModel.IsDevDriveCheckboxChecked = true;
     }
+
+    private void FilterSuggestions(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs args)
+    {
+        sender.ItemsSource = _searchFieldsAndValues[sender.Header.ToString()].Where(x => x.Contains(sender.Text));
+        return;
+    }
+
+    private async void SwitchToSearchPage(object sender, RoutedEventArgs e)
+    {
+        AddRepoViewModel.ChangeToSelectSearchTermsPage();
+        _searchFieldsAndValues.Clear();
+        ShowingSearchTermsGrid.Children.Clear();
+        GatheringSearchValuesGrid.Visibility = Visibility.Visible;
+        ShowingSearchTermsGrid.Visibility = Visibility.Collapsed;
+
+        var loginId = (string)AddRepoViewModel.SelectedAccount;
+        var searchTerms = AddRepoViewModel.GetSearchTerms();
+        ShowingSearchTermsGrid.RowSpacing = 10;
+
+        var searchTermRow = 0;
+        for (var termIndex = 0; termIndex < searchTerms.Count; termIndex++)
+        {
+            var localTermIndex = termIndex;
+            ShowingSearchTermsGrid.RowDefinitions.Add(new RowDefinition());
+
+            var searchFieldName = string.Empty;
+            var searchFieldSuggestions = await Task.Run(() => AddRepoViewModel.GetSuggestionsFor(loginId, new(), searchTerms[localTermIndex]));
+
+            _searchFieldsAndValues.Add(searchTerms[localTermIndex], searchFieldSuggestions);
+            var suggestBox = new AutoSuggestBox();
+            suggestBox.Header = searchTerms[localTermIndex];
+            suggestBox.ItemsSource = searchFieldSuggestions;
+            suggestBox.Text = searchFieldName;
+            suggestBox.TextChanged += FilterSuggestions;
+            ShowingSearchTermsGrid.Children.Add(suggestBox);
+            Grid.SetRow(suggestBox, searchTermRow++);

```

Comment:
Asking in case I missed something but is it possible for us to set these within the xaml itself?

Reply:
Very true.  In a previous iteration I had to terminate the loop early.  However, now that I'm using all the `searchTerms` this can go into the UI.  Added #2409 to track the work.